App.js

```diff
@@ -0,0 +1,171 @@
+      </div>
+    )
+  }
+}
+
+//add a new task
+class AddNewTask extends React.Component{
+  constructor(){
+    super();
+    this.state={
+      text: ""
+    }
+    this.InputChange = this.InputChange.bind(this);
+    this.SubmitNewTask = this.SubmitNewTask.bind(this);
+  }
+
+  InputChange(event){
+    this.setState({text: event.target.value});
+  }
+
+  SubmitNewTask(event){
+    this.props.GetNewTask(this.state.text);
+    this.setState({text: ""});
+  }
+
+
+  render(){
+    return (
+      <div className="AddNewTask-box">
+        <span className="AddNewTask-items">Add a new Task:</span>
+        <input placeholder="Type something about your new task..." className="AddNewTask-items" onChange={this.InputChange} value={this.state.text}/> 
+        <button className="AddNewTask-items" onClick={this.SubmitNewTask}>Add</button>
+      </div>
+      )
+  }
+}
+
+//task div
+class PageTasks extends React.Component{
+  constructor(){
+    super();
+    this.state={
+      newtasks:[],
+      donetasks:[]
+    }
+
+    this.AddNewTaskItem = this.AddNewTaskItem.bind(this);
+    this.SetTaskAsDone = this.SetTaskAsDone.bind(this);
+  }
+
+  AddNewTaskItem(text){
+    var addToTaskList = true;
+    this.state.newtasks.forEach(task => {
+      if(task.Content == text){
+        addToTaskList =  false;
+        return;
+      }
+    });
+    if(addToTaskList){
+      this.setState({newtasks: this.state.newtasks.concat({Content: text})});
+    }else{
+      alert("This task already exists!");
+    }
+    
+  }
+
+  SetTaskAsDone(text){
+    //update newTask list
+    var updateNewTaskList = [];
+    this.state.newtasks.forEach(task => {
+      if(task.Content != text){
+        updateNewTaskList.push(task);
+      }
+    });
+    this.setState({newtasks: updateNewTaskList});
+    //update doneTask list
+    this.setState({donetasks: this.state.donetasks.concat({Content: text})});
+  }
+
+  render(){
+    const taskFinishRate = Math.round(this.state.donetasks.length/(this.state.newtasks.length + this.state.donetasks.length) * 10000) / 100 + "%";
+    return (
+      <div>
+        <AddNewTask GetNewTask={this.AddNewTaskItem}/>
+        <div>
+    {(this.state.newtasks.length > 0 || this.state.donetasks.length > 0) && <h4 className="DisplayAllTasks-box">All tasks: (Task finish rate: <span>{taskFinishRate}</span>)</h4>}
+          <DisplayAllTasks TaskList={this.state.newtasks} SetTaskDone={this.SetTaskAsDone} TaskState="New"/>
+          <DisplayAllTasks TaskList={this.state.donetasks} TaskState="Done"/>
+        </div>
+      </div>
+    );
+  }
+}
+
+//show all tasks 
+class DisplayAllTasks extends React.Component{
+  render(){
+    const taskLists = [];
+    this.props.TaskList.forEach(item => {
+      console.log(item);
+      taskLists.push(<TaskItem content={item.Content} key={item.Content} state={this.props.TaskState} SetTaskDone={() => this.props.SetTaskDone(item.Content)} />)
+    });
```

Comment:
this `[].push()` inside of `.forEach()` can be replaced by using `.map()`. 

see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map